torch/distributed/fsdp/flat_param.py

```diff
@@ -792,6 +793,21 @@ def prepare_gradient(self):
                 # a GPU tensor (the new sharded gradient).
                 if not grad_offloaded:
                     flat_param._saved_grad_shard = flat_param.grad.data  # type: ignore[attr-defined]
+                    # If we're using mixed precision with keeping grads
+                    # casted, gradient here might still be of the reduced
+                    # dtype if we didn't clear / set the gradients to None
+                    # after previous forward. In that case, make sure
```

Comment:
"after previous forward"
Would I be correct to say this is more precisely "in the pre-backward"?

Reply:
I think this should actually just be "after previous backward", right? This code is running in pre-backwards?